Actions/DetermineDeploymentEnvironments/DetermineDeploymentEnvironments.ps1

```diff
@@ -46,6 +59,22 @@ function Get-BranchesFromPolicy($ghEnvironment) {
 . (Join-Path -Path $PSScriptRoot -ChildPath "..\AL-Go-Helper.ps1" -Resolve)
 
 $settings = $env:Settings | ConvertFrom-Json | ConvertTo-HashTable -recurse
+
+$generateALDocArtifact = ($type -eq 'Publish') -or $settings.ALDoc.ContinuousDeployment
+Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "GenerateALDocArtifact=$([int]$generateALDocArtifact)"
+Write-Host "GenerateALDocArtifact=$([int]$generateALDocArtifact)"
+
+$deployToGitHubPages = $settings.ALDoc.DeployToGitHubPages
+if ($deployToGitHubPages) {
+    $deployToGitHubPages = IsGitHubPagesAvailable
+}
+Add-Content -Encoding UTF8 -Path $env:GITHUB_OUTPUT -Value "DeployALDocArtifact=$([int]$deployToGitHubPages)"
+Write-Host "DeployALDocArtifact=$([int]$deployToGitHubPages)"
+
+if ($getEnvironments -eq '-') {
```

Comment:
DetermineDeploymentEnvironments is called during initialize to determine deployment environments and build the deploy matrix - I could have added ALDoc deployment to the matrix - or like now, add a special variable for whether or not DeployALDoc should be called.
Decided on the latter.

Reply:
My question was for `$getEnvironments`'s value of `-`. 

You cannot add ALDoc deployment to the deploy matrix, because the `Deploy` job does something completely different.

You want to use `DetermineDeploymentEnvironments` action in the new workflow, but it doesn't need to actually calculate all deployment environments. So, the `-` value of `$getEnvironments` is used to terminate the action and only output what's needed in the new workflow.

There should be a better design to separate the two calculations even if you prefer to keep them in the same action.

Reply:
Any suggestion to a different design?
2 actions?

Reply:
- Add separate action
- Change the action's logic to support 'github-pages' as input to `$getEnvironments`
- A parameter to the action for "types" of GitHub environments to retrieve. Where a type could be "for artifacts" or "for documentation".